src/v/redpanda/admin/server.cc

```diff
@@ -746,6 +746,11 @@ ss::future<ss::httpd::redirect_exception> admin_server::redirect_to_leader(
     // - if the host in the Host header matches one of our advertised kafka
     //   addresses, then assume that the peer's advertised kafka address
     //   with the same index will also be their public admin API address.
+    //   - a match in this case might be an exact match or the appearance
```

Comment:
The documentation isn't the most clear but it might be useful to add a link to [K8s' Documentation about Pod DNS](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields)?

Reply:
Hadn't seen this. Thanks! So if I understand correctly, this bit covers how we organize things in cloud, right?

> If there exists a headless Service in the same namespace as the Pod, with the same name as the subdomain, the cluster's DNS Server also returns A and/or AAAA records for the Pod's fully qualified hostname.

Reply:
_kinda_. The Pod DNS link does the best job of explaining why the logic you're implementing here could be useful but there's some missing links about how we get there.

The headless service is more of a detail in using StatefulSets, [which require a headless Service to exist](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#limitations). This isn't specific to cloud; it's generally applicable to anything deployed on a statefulset.